File: matlab/ch08/coolhxdat.m
% coolhxdat.m
% Physical properties
Cph = 0.92; Cpc = 1; rhoh = 59.76; rhoc = 59.87; kh = 0.3; kc = 0.36;
kw = 30; muh = 0.75; muc = 0.77; muwc = muc; muwh = muh;
% Operating conditions
mh = 6.2e4; ui = 5; T1 = 150; T2 = 120; t1 = 75; t2 = 100;
% Shell and tube
L = 15; Do = 0.0625; Di = 0.04017; Ds = 1.4375; Pt = 1/12; cl = 0.02083; Rfi = 0.004;
Rfo = 0;
% Parameters and basic properties
gc = 32.2; dTlm = ((T1-t2) - (T2-t1))/log((T1-t2)/(T2-t1)); % log-mean temp.
R = (T1-T2)/(t2-t1); S = (t2-t1)/(T1-t1);
F12den = (R-1)*log((2 - S*(R+1-sqrt(R^2+1)))/(2 - S*(R+1+sqrt(R^2+1))));
F12 = sqrt(R^2+1)*log((1-S)/(1-R*S))/F12den; % correction factor
Q = mh*Cph*(T1 - T2); mc = Q/(Cpc*(t2 - t1)); % Heat load and cold stream rate
Aci = mh/(rhoh*ui*3600); % tube-side total cross-sectional area
Nt = ceil(4*Aci/(pi*Di^2)); % number of tubes per pass
At = pi*Di*L; % heat transfer area per tube
dx = (Do - Di)/2; % tube thickness
Dlm = (Do - Di)/log(Do/Di); % log-mean diameter
De = (4/(pi*Do))*(Pt^2 - pi*Do^2/4); % hydraulic effective diameter
Ui = 150; % Assume Ui

File: python/ch08/coolhxdat.py
import numpy as np

def coolhxdat():
    """
    열교환기(Heat Exchanger) 관련 데이터를 정의하는 함수입니다.
    데이터를 딕셔너리 형태로 반환합니다.
    """
    # Physical properties
    data = {
        'Cph': 0.92, 'Cpc': 1.0, 'rhoh': 59.76, 'rhoc': 59.87, 
        'kh': 0.3, 'kc': 0.36, 'kw': 30.0, 'muh': 0.75, 
        'muc': 0.77, 'muwc': 0.77, 'muwh': 0.75,
        
        # Operating conditions
        'mh': 6.2e4, 'ui': 5.0, 'T1': 150.0, 'T2': 120.0, 
        't1': 75.0, 't2': 100.0,
        
        # Shell and tube
        'L': 15.0, 'Do': 0.0625, 'Di': 0.04017, 'Ds': 1.4375, 
        'Pt': 1/12, 'cl': 0.02083, 'Rfi': 0.004, 'Rfo': 0.0
    }
    
    # Parameters and basic properties (계산된 항목들)
    data['gc'] = 32.2
    
    # 로그 평균 온도차 (LMTD)
    dTlm = ((data['T1'] - data['t2']) - (data['T2'] - data['t1'])) / np.log((data['T1'] - data['t2']) / (data['T2'] - data['t1']))
    data['dTlm'] = dTlm
    
    # 보정 계수 (Correction factor) 계산
    R = (data['T1'] - data['T2']) / (data['t2'] - data['t1'])
    S = (data['t2'] - data['t1']) / (data['T1'] - data['t1'])
    
    F12den = (R - 1) * np.log((2 - S*(R + 1 - np.sqrt(R**2 + 1))) / (2 - S*(R + 1 + np.sqrt(R**2 + 1))))
    F12 = np.sqrt(R**2 + 1) * np.log((1 - S) / (1 - R*S)) / F12den
    data['F12'] = F12
    
    # 열 부하 및 냉각수 유량
    Q = data['mh'] * data['Cph'] * (data['T1'] - data['T2'])
    mc = Q / (data['Cpc'] * (data['t2'] - data['t1']))
    data['Q'] = Q
    data['mc'] = mc
    
    # 기하학적 파라미터 계산
    Aci = data['mh'] / (data['rhoh'] * data['ui'] * 3600)
    Nt = np.ceil(4 * Aci / (np.pi * data['Di']**2))
    At = np.pi * data['Di'] * data['L']
    
    data['Nt'] = Nt
    data['At'] = At
    data['dx'] = (data['Do'] - data['Di']) / 2
    data['Dlm'] = (data['Do'] - data['Di']) / np.log(data['Do'] / data['Di'])
    data['De'] = (4 / (np.pi * data['Do'])) * (data['Pt']**2 - np.pi * data['Do']**2 / 4)
    data['Ui'] = 150.0 # 가정값
    
    return data